### PR TITLE
Cc 32 customise court size

### PR DESCRIPTION
**Pull request by xunli253**
image

user can customise court by input width and length.

### CODE REVIEW

**Review by tatoMa — COMMENTED** [review]
Seems like the slider of the border is not working anymore.
image

**Review by tatoMa — COMMENTED** [review]
When clicking the color changer (the print icon) and then moving into the customise popup window, the icon is not reset back to the mouse default. It might be a little confusing.
 
image

**Review by tatoMa — COMMENTED** [review]
The error message is only showing after clicking the Set button. What if press the enter key?

Also, the width of the Set button is not as same as the Reset button.
image

**Comment by tatoMa** [comment]
> 
Or if you can show the error message after blur the input? (leave the input field)

**Comment by tatoMa** [comment]
> The error message is only showing after clicking the Set button. What if press the enter key?
> 
> Also, the width of the Set button is not as same as the Reset button. image

The gap between these 2 input fields is too big. The error message container has not enough space to show all the messages. Use different colors or outline style for the Reset button to better tell users that these 2 buttons are not sharing a similar functionality.

File: src/components/ProHalfCourt/CustomiseWindow.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        <Spacer />
+        <Icon as={isOpen ? BiChevronDown : BiChevronUp} color="white" width={6} height={6} />
+      </Flex>
+      <Collapse in={isOpen} animateOpacity>
+        <Flex
+          direction="column"
+          width="20vw"
+          padding={5}
+          border="1px solid #2C4E8A"
+          height={360}
```
Comment (Double-Twelve) [inline src/components/ProHalfCourt/CustomiseWindow.tsx:71]: why need to set the _height_?

File: src/components/ProHalfCourt/CustomiseWindow.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  FormHelperText,
+} from "@chakra-ui/react";
+import React, { useState } from "react";
+import { BiChevronDown, BiChevronUp, BiPencil } from "react-icons/bi";
+// inputWidth={clipWidth} setInputWidth={setClipWidth} inputLength = {clipLength} setInputLength={setClipLength
+const CustomiseWindow = ({ setInputWidth, setInputLength }: any) => {
+  const { isOpen, onToggle } = useDisclosure();
+
+  const [widthValue, setWidthValue] = useState("");
+  const [hightValue, setHightValue] = useState("");
```
Comment (Double-Twelve) [inline src/components/ProHalfCourt/CustomiseWindow.tsx:23]: 1. The type of the property should be changed to number
2. The property name is hightValue, but it corresponds to the length, please be consistent

File: src/components/ProHalfCourt/CustomiseWindow.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  Input,
+  Spacer,
+  useDisclosure,
+  Text,
+  ButtonGroup,
+  FormHelperText,
+} from "@chakra-ui/react";
+import React, { useState } from "react";
+import { BiChevronDown, BiChevronUp, BiPencil } from "react-icons/bi";
+// inputWidth={clipWidth} setInputWidth={setClipWidth} inputLength = {clipLength} setInputLength={setClipLength
```
Comment (Double-Twelve) [inline src/components/ProHalfCourt/CustomiseWindow.tsx:18]: what is the purpose of this comment?

File: src/store/reducer/courtSpecDataSlice.ts [hunk 4]
```diff
@@ -71,6 +75,14 @@ export const Slice = createSlice({
...
       return state;
     },
+    setNewCourtAreaXLength: (state: CourtSpec, action?: PayloadAction<number>) => {
+      state.activeCourt = { ...state.activeCourt, customizeCourtAreaXLength: action?.payload };
+      return state;
+    },
+    setNewCourtAreaYLength: (state: CourtSpec, action?: PayloadAction<number>) => {
+      state.activeCourt = { ...state.activeCourt, customizeCourtAreaYLength: action?.payload };
+      return state;
+    },
```
[the remark this answers is not in the document]
Comment (xunli253) [reply src/store/reducer/courtSpecDataSlice.ts]: fix that

File: src/components/ProHalfCourt/index.tsx [hunk 5]
```diff
@@ -1,6 +1,6 @@
-import { Stage, Layer, Group } from "react-konva";
-import { Flex } from "@chakra-ui/react";
-import { ReactReduxContext, Provider } from "react-redux";
+import { Stage, Layer, Group, Line } from "react-konva";
+import { Box, Flex } from "@chakra-ui/react";
```
[the remark this answers is not in the document]
Comment (xunli253) [reply src/components/ProHalfCourt/index.tsx]: fix it

File: src/components/ProHalfCourt/index.tsx [hunk 6]
```diff
@@ -12,13 +12,23 @@ import BorderDimension from "../BasketballCourt/BorderDimension";
...
+import CustomiseWindow from "./CustomiseWindow";
+import CustomiseCourtDimension from "./CustomiseCourtDimension";
+import { setNewCourtAreaYLength, setNewCourtAreaXLength } from "@/store/reducer/courtSpecDataSlice";
 
 const ProHalfCourt = () => {
+  const dispatch = useDispatch();
   const { courtAreaXLength, courtAreaYLength, borderLength, court, courtStartPoint } = useCourt();
   const ref = useRef<any>(null);
+  const [clipWidth, setClipWidth] = useState("");
+  const [clipLength, setClipLength] = useState("");
```
[the remark this answers is not in the document]
Comment (xunli253) [reply src/components/ProHalfCourt/index.tsx]: fix it

File: src/components/ProHalfCourt/CustomiseWindow.tsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  };
+  const resetHandle = () => {
+    setWidthValue("");
+    setHightValue("");
+    setInputWidth("");
+    setInputLength("");
+    setIsValid(true);
+  };
+  return (
+    <Box position="fixed" right={0} bottom="109px">
```
[the remark this answers is not in the document]
Comment (xunli253) [reply src/components/ProHalfCourt/CustomiseWindow.tsx]: fix it

File: src/components/ProHalfCourt/CustomiseWindow.tsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+        backgroundColor="#2C4E8A"
+        width="20vw"
+        height="5vh"
+        zIndex="2"
+        padding="5"
+        alignItems="center"
+        fontSize="xl"
+        onClick={onToggle}
+        cursor="pointer"
+        borderTopRadius={8}
```
[the remark this answers is not in the document]
Comment (xunli253) [reply src/components/ProHalfCourt/CustomiseWindow.tsx]: fix that

File: src/components/ProHalfCourt/CustomiseWindow.tsx (hunk 1 above)
Comment (xunli253) [reply src/components/ProHalfCourt/CustomiseWindow.tsx]: I set height to guarantee enough space for error information when users input invalid information.

File: src/components/ProHalfCourt/CustomiseWindow.tsx (hunk 2 above)
Comment (xunli253) [reply src/components/ProHalfCourt/CustomiseWindow.tsx]: I fix that.

File: src/components/ProHalfCourt/CustomiseWindow.tsx (hunk 3 above)
Comment (xunli253) [reply src/components/ProHalfCourt/CustomiseWindow.tsx]: Fix that.

**Review by tatoMa — COMMENTED** [review]
Do you think it looks nicer in this color schema?

image

image

Or the original one.
image

**Comment by xunli253** [comment]
> Do you think it looks nicer in this color schema?
> 
> image
> 
> image
> 
> Or the original one. image
I think the top one is nicer. I have updated my style. please check.

**Review by tatoMa — COMMENTED** [review]
Missed the style of hover state of the set button.
image

**Comment by xunli253** [comment]
> Missed the style of hover state of the set button. image

I have added variant of shareBtn for set button. It should have hover style now.

**Review by tatoMa — APPROVED** [review]
Nice work! The style of text "Customize" on the blueprint can look better with a few changes though but overall working well.

**Comment by Double-Twelve** [comment]
when I change the width or length of the court, the estimated tiles have no corresponding change

image

**Comment by xunli253** [comment]
> when I change the width or length of the court, the estimated tiles have no corresponding change
> 
> image

I  have fixed the issue. Please check

**Comment by Double-Twelve** [comment]
> > when I change the width or length of the court, the estimated tiles have no corresponding change
> > image
> 
> I have fixed the issue. Please check

Why tiles have black color?
image

**Comment by tatoMa** [comment]
> > > when I change the width or length of the court, the estimated tiles have no corresponding change
> > > image
> > 
> > 
> > I have fixed the issue. Please check
> 
> Why tiles have black color? image

It is might be related to the old tile color pattern calculator. It is not an easy one and has a few known bugs.

**Comment by xunli253** [comment]
> > > > when I change the width or length of the court, the estimated tiles have no corresponding change
> > > > image
> > > 
> > > 
> > > I have fixed the issue. Please check
> > 
> > 
> > Why tiles have black color? image
> 
> It is might be related to the old tile color pattern calculator. It is not an easy one and has a few known bugs.
So, do I need to open new card to fix this one?

**Review by GLC-coder — APPROVED** [review]
Good job, Lee. LGTM

**Pull request merged**